src/esphomelib/sensor/tx20_component.cpp

```diff
@@ -0,0 +1,172 @@
+#include "esphomelib/defines.h"
+
+#ifdef USE_TX20
+
+#include "esphomelib/espmath.h"
+#include "esphomelib/log.h"
+#include "esphomelib/sensor/tx20_component.h"
+
+ESPHOMELIB_NAMESPACE_BEGIN
+
+namespace sensor {  // namespace
+
+static const char *TAG = "sensor.tx20";
+static const uint8_t MAX_BUFFER_SIZE = 41;
+static const uint16_t TX20_MAX_TIME = MAX_BUFFER_SIZE * 1220 + 5000;
+static const uint16_t TX20_BIT_TIME = 1200;
+static const std::string DIRECTIONS[] = {"N", "NNE", "NE", "ENE", "E", "ESE", "SE", "SSE",
+                                         "S", "SSW", "SW", "WSW", "W", "WNW", "NW", "NNW"};
```

Comment:
Remove `DIRECTIONS` now that there's no text sensor. Also `std::string` has the same issue as `std::vector<>`, it prevents compiler optimizations - it would have needed to be `static const char *DIRECTIONS[]`